File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn upper-case-character?
  [char]
  (and (java.lang.Character/isLetter char)
       (java.lang.Character/isUpperCase char)))

(defn all-upper-case?
  [s]
  (let [alpha-only (str/replace s #"[^\p{Alpha}]" "")]
    (and (seq alpha-only)
         (every? upper-case-character? alpha-only))))

(defn response-for [s]
  (let [normalized-input (str/trim s)
        silence? (= "" normalized-input)
        question? (and (not silence?)
                       (str/ends-with? normalized-input "?"))
        yelling? (and (not silence?)
                      (all-upper-case? normalized-input))]
    (cond
      (and yelling? question?) "Calm down, I know what I'm doing!"
      question? "Sure."
      yelling? "Whoa, chill out!"
      silence? "Fine. Be that way!"
      :else "Whatever.")))

File: python/bob/bob.py
def response(hey_bob):
    normalized = hey_bob.strip()
    is_silence = normalized == ""
    is_question = not is_silence and normalized.endswith("?")
    is_yelling = not is_silence and normalized.isupper()

    if is_question and is_yelling:
        return "Calm down, I know what I'm doing!"
    if is_question:
        return "Sure."
    if is_yelling:
        return "Whoa, chill out!"
    if is_silence:
        return "Fine. Be that way!"
    return "Whatever."
